src/signer.rs

```diff
@@ -167,38 +181,33 @@ where
         }
 
         let mut balances = BalanceAdapter::<T>::new();
-        for (address, ms_data) in self.multisig_info.iter() {
-            let account = Pallet::<T>::to_native_account(address)?;
+        for (account, ms_data) in self.multisig_info.iter() {
             balances
-                .write_cheque(&account, &BalanceOf::<T>::from(ms_data.cheque_limit))
+                .write_cheque(account, &BalanceOf::<T>::from(ms_data.cheque_limit))
                 .map_err(|_| Error::<T>::InsufficientBalance)?;
         }
 
         Ok(balances)
     }
 
-    /// Consumes `ScriptSignatureHandler` and returns innner `Multisig`.
-    pub(crate) fn into_inner(self) -> Multisig<T::MaxScriptSigners> {
+    /// Consumes [`ScriptSignatureHandler`] and returns innner `Multisig`.
+    pub(crate) fn into_inner(self) -> Multisig<T::AccountId, T::MaxScriptSigners> {
         self.multisig_info
     }
 
-    /// Consumes `ScriptSignatureHandler` and returns accounts of all signers as vector.
+    /// Consumes [`ScriptSignatureHandler`] and returns accounts of all signers as vector.
```

Comment:
Maybe we don't need to return `Result` anymore but dump the vector of accounts directly